src/AppBundle/Event/Listener/RenderMenuListener.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace AppBundle\Event\Listener;
+
+use AppBundle\User\UserGroups;
+use EzSystems\EzPlatformAdminUi\Menu\Event\ConfigureMenuEvent;
+use Knp\Menu\ItemInterface;
+use Knp\Menu\Util\MenuManipulator;
+use Symfony\Component\Translation\TranslatorInterface;
+
+class RenderMenuListener
+{
+    private const PERSONALIZATION_MENU_ITEM = 'personalization';
+    private const EZTAGS_MENU_ITEM = 'eztags';
+
+    /** @var \AppBundle\User\UserGroups */
+    private $userGroups;
+
+    /** @var \Symfony\Component\Translation\TranslatorInterface */
+    private $translator;
+
+    /**  @var int */
+    private $administratorUsersLocationId;
+
+    /** @var string */
+    private $personalizationLicenseKey;
+
+    /** @var int */
+    private $personalizationCustomerId;
+
+    /**
+     * @param \AppBundle\User\UserGroups $userGroups
+     * @param \Symfony\Component\Translation\TranslatorInterface $translator
+     * @param int $administratorUsersLocationId
+     * @param int|null $personalizationCustomerId
+     * @param string|null $personalizationLicenseKey
+     */
+    public function __construct(
+        UserGroups $userGroups,
+        TranslatorInterface $translator,
+        int $administratorUsersLocationId,
+        ?int $personalizationCustomerId,
+        ?string $personalizationLicenseKey
+    ) {
+        $this->userGroups = $userGroups;
+        $this->translator = $translator;
+        $this->administratorUsersLocationId = $administratorUsersLocationId;
+        $this->personalizationCustomerId = $personalizationCustomerId;
+        $this->personalizationLicenseKey = $personalizationLicenseKey;
+    }
+
+    /** @param \EzSystems\EzPlatformAdminUi\Menu\Event\ConfigureMenuEvent $event */
+    public function renderMenu(ConfigureMenuEvent $event): void
+    {
+        $menu = $event->getMenu();
+        $manipulator = new MenuManipulator();
+
+        $this->configureEzTags($menu, $manipulator);
+        $this->configureRecommendation($menu, $manipulator);
+    }
+
+    /**
+     * @param \Knp\Menu\ItemInterface $item
+     * @param \Knp\Menu\Util\MenuManipulator $menuManipulator
+     */
+    private function configureEzTags(ItemInterface $item, MenuManipulator $menuManipulator): void
+    {
+        if ($item->getChild(self::EZTAGS_MENU_ITEM)) {
+            $menuManipulator->moveToLastPosition($item[self::EZTAGS_MENU_ITEM]);
+        }
+    }
+
+    /**
+     * @param \Knp\Menu\ItemInterface $item
+     * @param \Knp\Menu\Util\MenuManipulator $menuManipulator
+     */
+    private function configureRecommendation(ItemInterface $item, MenuManipulator $menuManipulator): void
+    {
+        if ($this->isAdmin() && $this->hasRecommendationCredentials()) {
```

Comment:
Not a big fan of this approach (only administrators can view the menu entry).
I would rather use a custom policy and it will be a good example in the demo
https://doc.ezplatform.com/en/latest/guide/custom_policies/#toc

If you change for this approach, then you won't have to use the parameters in yml. Especially because you are using a paramater implemented for a specific feature (premium content).

Reply:
I've added custom policy. @SylvainGuittard could you review once again please?